### PR TITLE
[AMORO-1794] Add cache catalog config for iceberg catalog

### LINKED ISSUE #1794: [Improvement]: Use `CachingCatalog` to reduce the time cost of `IcebergCatalogWrapper#loadTable`
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Every time ArcticTable is obtained, it will go through a complete load table process(org.apache.iceberg.catalog.Catalog#loadTable). We can cache the loaded table to reduce the time cost of get ArcticTable.

### How should we improve?

import `org.apache.iceberg.CachingCatalog`

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhongqishang**
## Why are the changes needed?

Fix #1794 

## Brief change log

  - *Use CachingCatalog wrap caltalog*
  - *Add class IcebergCachingCatalogWrapper.java*

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by baiyangtx** [comment]
Maybe this PR could be closed due to this PR https://github.com/NetEase/amoro/pull/2298

**Comment by zhongqishang** [comment]
> Maybe this PR could be closed due to this PR #2298

@baiyangtx 
In #2298, It seems that cache support for the native iceberg format catalog is missing, I have added it in the this PR.

PTAL.

File: core/src/main/java/com/netease/arctic/formats/iceberg/IcebergCatalog.java [hunk 1]
```diff
@@ -21,38 +21,39 @@
...
   @Override
   public List<String> listDatabases() {
-    if (icebergCatalog instanceof SupportsNamespaces) {
-      return ((SupportsNamespaces) icebergCatalog)
-          .listNamespaces().stream().map(ns -> ns.level(0)).collect(Collectors.toList());
-    }
-    return Lists.newArrayList();
+    return asNamespaceCatalog().listNamespaces().stream()
+        .map(ns -> ns.level(0))
+        .collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (zhongqishang) [reply core/src/main/java/com/netease/arctic/formats/iceberg/IcebergCatalog.java:56]: Similarly, the behavior of `createDatabase`, `dropDatabase` also has changed when iceberg catalog does't support namsespace.

I consider :
1. Be consistent with `BasicMixedIcebergCatalog` behavior;
2. I checked that `HiveCatalog`, `RESTCatalog`, `JdbcCatalog`, `InMemoryCatalog`, `HadoopCatalog`, `GlueCatalog`, and `DynamoDbCatalog` all have `SupportsNamespaces` implemented, so the change in behavior will not have an impact.

So, For iceberg catalog does't support namsespace, throw an exception may be a better way.

WDYT?

**Comment by baiyangtx** [comment]
@zhongqishang There is a conflict with master branch, could you fix that?

**Review by baiyangtx — APPROVED** [review]
LGTM, thanks for your contribution.
